RiscVPkg/Include/IndustryStandard/RiscVOpensbi.h

```diff
@@ -102,14 +102,26 @@
 
 #define RISC_V_MAX_HART_SUPPORTED 16
 
+typedef
+VOID
+(EFIAPI *RISCV_HART_SWITCH_MODE)(
+  IN  UINTN   Arg0,
+  IN  UINTN   Arg1,
+  IN  UINTN   NextAddr,
+  IN  UINTN   NextMode,
+  IN  BOOLEAN NextVirt
+  );
+
+
 //
 // keep the structure member in 64-bit alignment.
 //
 typedef struct {
-    UINT64          IsaExtensionSupported;  // The ISA extension this core supported.
-    RISCV_UINT128   MachineVendorId;        // Machine vendor ID
-    RISCV_UINT128   MachineArchId;          // Machine Architecture ID
-    RISCV_UINT128   MachineImplId;          // Machine Implementation ID
+    UINT64                 IsaExtensionSupported;  // The ISA extension this core supported.
+    RISCV_UINT128          MachineVendorId;        // Machine vendor ID
+    RISCV_UINT128          MachineArchId;          // Machine Architecture ID
+    RISCV_UINT128          MachineImplId;          // Machine Implementation ID
+    RISCV_HART_SWITCH_MODE HartSwitchMode;

```

Comment:
It would be better to create another structure for internal functions if consider more functions would be public.

Reply:
Good idea

Reply:
I can't quite figure out how to do that. I'm not sure where to put this structure in `RiscVOpenSbiHartSwitchMode` and why it needs to be 64 byte aligned.

Reply:
Just something like this

typedef struct {
    UINT64          IsaExtensionSupported;  // The ISA extension this core supported.
    RISCV_UINT128   MachineVendorId;        // Machine vendor ID
    RISCV_UINT128   MachineArchId;          // Machine Architecture ID
    RISCV_UINT128   MachineImplId;          // Machine Implementation ID
   **EDK2_OPEN_SBI_INTERNAL_FUNCTION OpensbiInternalFunc;**
} EFI_RISCV_FIRMWARE_CONTEXT_HART_SPECIFIC;

typedef struct {
    RISCV_HART_SWITCH_MODE HartSwitchMode;
   ...
   ...
}    EDK2_OPEN_SBI_INTERNAL_FUNCTION


64-bit alignment for load/store using 64-bit regs is the performance consideration. and some hardware may not support misaligned access.